RealmSwift/MongoClient.swift

```diff
@@ -483,4 +502,230 @@ extension MongoCollection {
+    /// - Returns: A ChangeStream which will manage the streaming events.
+    public func watch(matchFilter: Document, delegate: ChangeEventDelegate, queue: DispatchQueue = .main) -> ChangeStream {
+        let filterBSON = ObjectiveCSupport.convert(object: .document(matchFilter)) as! [String: RLMBSON]
+        return self.__watch(withMatchFilter: filterBSON,
+                            delegate: ChangeEventDelegateProxy(delegate),
+                            delegateQueue: queue)
+    }
+
+    /// Opens a MongoDB change stream against the collection to watch for changes
+    /// made to specific documents. The documents to watch must be explicitly
+    /// specified by their _id.
+    /// - Parameters:
+    ///   - filterIds: The list of _ids in the collection to watch.
+    ///   - delegate: The delegate that will react to events and errors from the resulting change stream.
+    ///   - queue: Dispatches streaming events to an optional queue, if no queue is provided the main queue is used
+    /// - Returns: A ChangeStream which will manage the streaming events.
+    public func watch(filterIds: [ObjectId], delegate: ChangeEventDelegate, queue: DispatchQueue = .main) -> ChangeStream {
+        let filterBSON = ObjectiveCSupport.convert(object: .array(filterIds.map {AnyBSON($0)})) as! [RLMObjectId]
+        return self.__watch(withFilterIds: filterBSON,
+                            delegate: ChangeEventDelegateProxy(delegate),
+                            delegateQueue: queue)
+    }
+}
+
+private class ChangeEventDelegateProxy: RLMChangeEventDelegate {
+
+    private weak var proxyDelegate: ChangeEventDelegate?
+
+    init(_ proxyDelegate: ChangeEventDelegate) {
+        self.proxyDelegate = proxyDelegate
+    }
+
+    func changeStreamDidOpen(_ changeStream: RLMChangeStream) {
+        proxyDelegate?.changeStreamDidOpen(changeStream)
+    }
+
+    func changeStreamDidCloseWithError(_ error: Error?) {
+        proxyDelegate?.changeStreamDidClose(with: error)
+    }
+
+    func changeStreamDidReceiveError(_ error: Error) {
+        proxyDelegate?.changeStreamDidReceive(error: error)
+    }
+
+    func changeStreamDidReceiveChangeEvent(_ changeEvent: RLMBSON) {
+        let bson = ObjectiveCSupport.convert(object: changeEvent)
+        proxyDelegate?.changeStreamDidReceive(changeEvent: bson)
+    }
 }
+
+#if canImport(Combine)
+import Combine
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+extension Publishers {
+
+    class WatchSubscription<S: Subscriber>: ChangeEventDelegate, Subscription where S.Input == AnyBSON, S.Failure == Error {
+
+        private let collection: MongoCollection
+        private var changeStream: ChangeStream?
+        private var subscriber: S?
+
+        init(collection: MongoCollection, subscriber: S, queue: DispatchQueue = .main) {
```

Comment:
This probably just wants to be one initializer that has some optional parameters? Checking which parameters were actually supplied in WatchPublisher rather than here is just adding some extra boilerplate.